hello-world/simponic/Makefile

```
@@ -0,0 +1,29 @@
+.DEFAULT_GOAL := os.iso
+
+OBJECTS=entry.o kernel.o
+KERNEL_BUILD_FLAGS=-std=c99 -ffreestanding -O0 -Wall -Wextra
+OS_BUILD_FLAGS=-ffreestanding -O2 -nostdlib -lgcc
+
+entry.o: boot.asm 
+	nasm -felf32 boot.asm -o entry.o
+
+kernel.o: kernel.c
+	i386-elf-gcc -c kernel.c -o kernel.o $(KERNEL_BUILD_FLAGS)
+
+os.bin: kernel.o entry.o
+	i386-elf-gcc -T linker.ld -o os.bin $(OS_BUILD_FLAGS) $(OBJECTS)
+
+os.iso: os.bin
+	mkdir -p iso/boot/grub
+	cp os.bin iso/boot/os.bin
+	printf "menuentry 'HelloOS' {\n  multiboot /boot/os.bin\n  set gfxpayload=text\n}" > iso/boot/grub/grub.cfg
+	grub-mkrescue -o os.iso iso
+
+demo:
+	qemu-system-i386 demo/os.iso
+
```

Sorry for so many back-and-forths; I just caught this:

There is an oddity here with `make`. If your `make` target is the same as the build directory, make tries to do some clever work.

Basically, *if the build target is a directory which already exists, and has no new dependencies to build, make will not execute the target*

You could rename the target or the build directory, but the typical solution to this would be to add a `.PHONY` target. Here you can name all the targets which should be run on explicit call.

```suggestion
demo:
	qemu-system-i386 demo/os.iso

.PHONY: run demo
```